libhif/hif-context.c

```diff
@@ -140,6 +142,7 @@ hif_context_init(HifContext *context)
 {
     HifContextPrivate *priv = GET_PRIVATE(context);
     priv->install_root = g_strdup("/");
+    priv->source_root = g_strdup("/");
```

Comment:
I would default to sourceroot == installroot.  So set `source_root = NULL` here, then in the accessor, do `return priv->source_root ? priv->source_root : priv->install_root` ?